File: bottom-navigation/src/main/java/it/sephiroth/android/library/bottomnavigation/ItemsLayoutContainer.java
package it.sephiroth.android.library.bottomnavigation;

import android.support.annotation.IdRes;
import android.support.annotation.NonNull;
import android.view.View;
import android.view.ViewGroup;

import proguard.annotation.Keep;
import proguard.annotation.KeepClassMembers;

/**
 * Created by alessandro on 4/4/16 at 11:13 PM.
 * Project: Material-BottomNavigation
 */
@Keep
@KeepClassMembers
public interface ItemsLayoutContainer {
    void setSelectedIndex(int index, final boolean animate);

    int getSelectedIndex();

    void populate(@NonNull MenuParser.Menu menu);

    void setLayoutParams(ViewGroup.LayoutParams params);

    void setOnItemClickListener(OnItemClickListener listener);

    View findViewById(@IdRes final int id);

    void removeAll();
}

File: bottom-navigation/src/main/java/it/sephiroth/android/library/bottomnavigation/ItemsLayoutContainer.kt
package it.sephiroth.android.library.bottomnavigation

import android.content.Context
import android.view.ViewGroup

/**
 * Created by alessandro on 4/4/16 at 11:13 PM.
 * Project: Material-BottomNavigation
 */
abstract class ItemsLayoutContainer(context: Context) : ViewGroup(context) {

    abstract fun getSelectedIndex(): Int

    abstract fun setSelectedIndex(index: Int, animate: Boolean)

    abstract fun populate(menu: MenuParser.Menu)

    var itemClickListener: OnItemClickListener? = null

    abstract fun removeAll()

    abstract fun setItemEnabled(index: Int, enabled: Boolean)
}
